aten/src/ATen/cuda/PeerToPeerAccess.cpp

```diff
@@ -58,4 +65,121 @@ bool get_p2p_access(c10::DeviceIndex dev, c10::DeviceIndex dev_to_access) {
   return cache;
 }
 
-}  // namespace at::cuda::detail
+namespace {
+nvmlDevice_t get_nvml_device(c10::DeviceIndex dev) {
+#ifdef PYTORCH_C10_DRIVER_API_SUPPORTED
+  static bool nvml_init [[maybe_unused]] = []() {
+    TORCH_INTERNAL_ASSERT(NVML_SUCCESS == DriverAPI::get()->nvmlInit_v2_());
+    return true;
+  }();
+
+  auto prop = at::cuda::getDeviceProperties(dev);
+  char pci_id // NOLINT(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
+      [NVML_DEVICE_PCI_BUS_ID_BUFFER_SIZE];
+  snprintf(
```

Comment:
Use fmt::printf here. It's more efficient and safe. Use fmt::streamed for any objects that have custom ostream casters that you don't want to point to fmt.